src/store/RelayRecordStore.js

```diff
@@ -112,10 +113,11 @@ class RelayRecordStore {
   /**
    * Get the data ID associated with a storage key (and optionally an
    * identifying argument value) for a root query.
+   * toDO ?any should be something like number|string|object
    */
   getDataID(
     storageKey: string,
-    identifyingArgValue: ?string
+    identifyingArgValue: ?any
```

Comment:
Let's use `identifyingArgValue: mixed` here (no need for the `?` optional annotation since mixed includes undefined/null). With `any` Flow will stop doing inference on these values which will reduce the chance of catching bugs.